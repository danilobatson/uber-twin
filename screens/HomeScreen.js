import { StyleSheet, Text, SafeAreaView, View, Image } from 'react-native';
import React from 'react';
import tw from 'tailwind-react-native-classnames';

const HomeScreen = () => {
  return (
    <SafeAreaView style={tw`bg-white h-full`}>
      <View style={tw`p-5`}>
              <Image
                  accessibilityLabel='Uber Logo'
          source={{
            uri: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/58/Uber_logo_2018.svg/2560px-Uber_logo_2018.svg.png',
          }}
          style={{ width: 100, height: 100, resizeMode: 'contain' }}
        />
      </View>
    </SafeAreaView>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({});
