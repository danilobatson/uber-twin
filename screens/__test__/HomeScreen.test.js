import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react-native';

import HomeScreen from '../HomeScreen';
import { before, beforeEach } from 'node:test';

describe('<HomeScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Uber logo has accessible label', () => {
    render(<HomeScreen />);
    expect(screen.getByLabelText('Uber Logo')).toBeTruthy();
  });
});
