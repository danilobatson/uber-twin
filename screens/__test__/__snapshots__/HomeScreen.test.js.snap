// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <Image
      accessibilityLabel="Uber Logo"
      source={
        {
          "uri": "https://upload.wikimedia.org/wikipedia/commons/thumb/5/58/Uber_logo_2018.svg/2560px-Uber_logo_2018.svg.png",
        }
      }
      style={
        {
          "height": 100,
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
